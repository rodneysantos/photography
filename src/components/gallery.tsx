import { For } from "solid-js";
import { Photo } from "../models";

interface GalleryProps {
	photos: Photo[];
}

export const Gallery = (props: GalleryProps) => {
	return (
		<div class="flex justify-center">
			<div
				data-testid="gallery-list"
				class="container grid grid-cols-2 gap-12 3xl:auto-rows-3xl 4xl:auto-rows-4xl 5xl:auto-rows-5xl"
			>
				<For each={props.photos}>
					{(photo) => {
						return (
							<img
								class="object-cover object-center w-full h-full aspect-square"
								src={photo.url}
								alt=""
							/>
						);
					}}
				</For>
			</div>
		</div>
	);
};
