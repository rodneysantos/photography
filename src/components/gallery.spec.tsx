import { cleanup, render } from "@solidjs/testing-library";
import { afterEach, describe, expect, it } from "vitest";
import { Gallery } from "./gallery";

describe("gallery", () => {
	afterEach(cleanup);

	it("renders correctly", () => {
		// act
		const { asFragment } = render(() => <Gallery photos={[]} />);

		// assert
		expect(asFragment()).toMatchSnapshot();
	});

	it("renders the photos", () => {
		// arrange
		const photos = [
			{ id: 1, url: "photo1" },
			{ id: 2, url: "photo2" },
			{ id: 3, url: "photo3" },
		];

		// act
		const { getAllByAltText } = render(() => <Gallery photos={photos} />);

		// assert
		expect(getAllByAltText("")).toHaveLength(3);
	});
});
