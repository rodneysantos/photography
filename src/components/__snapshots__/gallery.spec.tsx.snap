// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`gallery > renders correctly 1`] = `"<div class=\\"flex justify-center\\"><div data-testid=\\"gallery-list\\" class=\\"container grid grid-cols-2 gap-12 3xl:auto-rows-3xl 4xl:auto-rows-4xl 5xl:auto-rows-5xl\\"></div></div>"`;
